Amazon.QLDB.Driver/driver/QldbDriver.cs

```diff
@@ -141,9 +146,23 @@ public T Execute<T>(Func<TransactionExecutor, T> func)
         /// <exception cref="AmazonServiceException">Thrown when there is an error executing against QLDB.</exception>
         public T Execute<T>(Func<TransactionExecutor, T> func, Action<int> retryAction)
         {
-            using (var session = this.sessionPool.GetSession())
+            while (true)
```

Comment:
So the problem of this is that the `RetryLimit` will be ignored when there is an InvalidSessionException or a TransactionAlreadyOpenException. Ideally, what we want is to consume a retry attempt when this exceptions are thrown.

Reply:
Yes, I noticed that. I was just trying to make it consistent with Java for now. When we change the retry logic, we can improve this behavior.

Reply:
TransactionAlreadyOpenException utilizes RetryLimit https://github.com/awslabs/amazon-qldb-driver-dotnet/pull/25/files#diff-70551e8ff5c2af7ba81186c2a58c35b0L171-L173

Reply:
I checked our data is super rare that this might happen. we can move on with this change